plugins/woocommerce/client/blocks/assets/js/blocks/product-reviews/inner-blocks/reviews-pagination/edit.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { PanelBody } from '@wordpress/components';
+import type { BlockEditProps } from '@wordpress/blocks';
+import {
+	InspectorControls,
+	useBlockProps,
+	// @ts-expect-error useInnerBlocksProps is not exported from @wordpress/block-editor
+	useInnerBlocksProps,
+	store as blockEditorStore,
+	Warning,
+} from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { ReviewsPaginationArrowControls } from './reviews-pagination-arrow-controls';
+import './editor.scss';
+
+const TEMPLATE = [
+	[ 'woocommerce/product-reviews-pagination-previous' ],
+	[ 'woocommerce/product-reviews-pagination-numbers' ],
+	[ 'woocommerce/product-reviews-pagination-next' ],
+];
+
+type Props = BlockEditProps< { paginationArrow: string | number | undefined } >;
+
+export default function Edit( {
+	attributes: { paginationArrow },
+	setAttributes,
+	clientId,
+}: Props ) {
+	const hasNextPreviousBlocks = useSelect(
+		( select ) => {
+			const { getBlocks } = select( blockEditorStore );
```

Comment:
The type for `@wordpress/block-editor`'s store is outdated. It's typed properly in more recent version. Because of that, I decided this as is without adding any comment. The warning will be solved automatically when we update the type packages.